### ISSUE TITLE
Avalonia does not work in Windows 10 on Mac

### ISSUE THREAD

**Issue by nopara73** [AvaloniaUI/Avalonia#1940]
### General Description

`libSkiaSharp` fails to load at `dotnet run` on the `ControlCatalog.NetCore` sample project with Windows 10, when the operating system is installed on a Mac through [Boot Camp.](https://support.apple.com/boot-camp)

### How To Reproduce?

1. Get a Mac, install Windows 10 through Boot Camp.
2. `git clone https://github.com/AvaloniaUI/Avalonia.git --recursive`
3. `cd Avalonia\samples\ControlCatalog.NetCore`
4. `dotnet run`

### Environment

I was able to reliably reproduce this issue consistently on 3 computers with Boot Camp setup, of which only one of it I have the specs for: Windows 10 Home, x64, running it through Boot Camp on a MacBook Pro, 2017 version.

@trevorwidger also has this issue, I know his environment is: Windows 10 Professional, x64, running it through Boot Camp on a Mac.  

@manakill also has this issue, I don't know his spec.

### Logs

```sh
C:\Users\Bali\Desktop\Avalonia\samples\ControlCatalog.NetCore>dotnet run
MenuItem.cs(424,40): warning CS1574: XML comment has cref attribute 'Header' that could not be resolved [C:\Users\Bali\Desktop\Avalonia\src\Avalonia.Controls\Avalonia.Controls.csproj]
DrawingContextImpl.cs(390,26): warning CS1572: XML comment has a param tag for 'interpolationMode', but there is no parameter by that name [C:\Users\Bali\Desktop\Avalonia\src\Skia\Avalonia.Skia\Avalonia.Skia.csproj]
SurfaceRenderTarget.cs(52,58): warning CS1573: Parameter 'gpu' has no matching param tag in the XML comment for 'SurfaceRenderTarget.CreateSurface(GRContext, int, int, PixelFormat?)' (but other parameters do) [C:\Users\Bali\Desktop\Avalonia\src\Skia\Avalonia.Skia\Avalonia.Skia.csproj]
Converters\SetterValueTypeConverter.cs(26,20): warning CS0618: 'SetterValueTypeConverter.ConvertSetterValue(ITypeDescriptorContext, XamlSchemaContext, CultureInfo, Setter, object)' is obsolete: 'TODO: try assosiate Setter.Value property with SetterValueTypeConverter, so far coouldn't make it :(' [C:\Users\Bali\Desktop\Avalonia\src\Markup\Avalonia.Markup.Xaml\Avalonia.Markup.Xaml.csproj]
PortableXaml\AvaloniaXamlSchemaContext.cs(90,24): warning CS0618: 'AvaloniaXamlSchemaContext.ResolveSimpleTypeName(string, string)' is obsolete: 'TODO: remove once it's fixed in Portable.xaml' [C:\Users\Bali\Desktop\Avalonia\src\Markup\Avalonia.Markup.Xaml\Avalonia.Markup.Xaml.csproj]
PortableXaml\AvaloniaXamlType.cs(229,29): warning CS0618: 'SetterValueTypeConverter.ConvertSetterValue(ITypeDescriptorContext, XamlSchemaContext, CultureInfo, Setter, object)' is obsolete: 'TODO: try assosiate Setter.Value property with SetterValueTypeConverter, so far coouldn't make it :(' [C:\Users\Bali\Desktop\Avalonia\src\Markup\Avalonia.Markup.Xaml\Avalonia.Markup.Xaml.csproj]

Unhandled Exception: System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. ---> System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. ---> System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. ---> Portable.Xaml.XamlObjectWriterException: Could not convert object 'M 0,0 c 0,0 50,0 50,-50 c 0,0 50,0 50,50 h -50 v 50 l -50,-50 Z' (of type System.String) to {clr-namespace:Avalonia.Media;assembly=Avalonia.Visuals}Geometry: Unable to load DLL 'libSkiaSharp' or one of its dependencies: The specified module could not be found. (Exception from HRESULT: 0x8007007E) ---> System.DllNotFoundException: Unable to load DLL 'libSkiaSharp' or one of its dependencies: The specified module could not be found. (Exception from HRESULT: 0x8007007E)
   at SkiaSharp.SkiaApi.sk_path_new()
   at SkiaSharp.SKPath..ctor()
   at Avalonia.Skia.StreamGeometryImpl.CreateEmptyPath() in C:\Users\Bali\Desktop\Avalonia\src\Skia\Avalonia.Skia\StreamGeometryImpl.cs:line 68
   at Avalonia.Skia.StreamGeometryImpl..ctor() in C:\Users\Bali\Desktop\Avalonia\src\Skia\Avalonia.Skia\StreamGeometryImpl.cs:line 40
   at Avalonia.Skia.PlatformRenderInterface.CreateStreamGeometry() in C:\Users\Bali\Desktop\Avalonia\src\Skia\Avalonia.Skia\PlatformRenderInterface.cs:line 51
   at Avalonia.Media.StreamGeometry.CreateDefiningGeometry() in C:\Users\Bali\Desktop\Avalonia\src\Avalonia.Visuals\Media\StreamGeometry.cs:line 72
   at Avalonia.Media.Geometry.get_PlatformImpl() in C:\Users\Bali\Desktop\Avalonia\src\Avalonia.Visuals\Media\Geometry.cs:line 47
   at Avalonia.Media.StreamGeometry.Open() in C:\Users\Bali\Desktop\Avalonia\src\Avalonia.Visuals\Media\StreamGeometry.cs:line 63
   at Avalonia.Media.StreamGeometry.Parse(String s) in C:\Users\Bali\Desktop\Avalonia\src\Avalonia.Visuals\Media\StreamGeometry.cs:line 40
   at Avalonia.Media.Geometry.Parse(String s) in C:\Users\Bali\Desktop\Avalonia\src\Avalonia.Visuals\Media\Geometry.cs:line 77
   at Avalonia.Markup.Xaml.Converters.ParseTypeConverter`1.ConvertFrom(ITypeDescriptorContext context, CultureInfo culture, Object value) in C:\Users\Bali\Desktop\Avalonia\src\Markup\Avalonia.Markup.Xaml\Converters\ParseTypeConverter.cs:line 68
   at Portable.Xaml.XamlObjectWriterInternal.GetCorrectlyTypedValue(XamlMember xm, XamlType xt, Object value) in C:\Users\Bali\Desktop\Avalonia\src\Markup\Avalonia.Markup.Xaml\PortableXaml\portable.xaml.github\src\Portable.Xaml\Portable.Xaml\XamlObjectWriter.cs:line 691
   --- End of inner exception stack trace ---
   at Portable.Xaml.XamlObjectWriterInternal.GetCorrectlyTypedValue(XamlMember xm, XamlType xt, Object value) in C:\Users\Bali\Desktop\Avalonia\src\Markup\Avalonia.Markup.Xaml\PortableXaml\portable.xaml.github\src\Portable.Xaml\Portable.Xaml\XamlObjectWriter.cs:line 698
   at Portable.Xaml.XamlObjectWriterInternal.StoreAppropriatelyTypedValue(ObjectState state, MemberAndValue ms, Object obj, Object keyObj) in C:\Users\Bali\Desktop\Avalonia\src\Markup\Avalonia.Markup.Xaml\PortableXaml\portable.xaml.github\src\Portable.Xaml\Portable.Xaml\XamlObjectWriter.cs:line 626
   at Portable.Xaml.XamlObjectWriterInternal.StoreAppropriatelyTypedValue(Object obj, Object keyObj) in C:\Users\Bali\Desktop\Avalonia\src\Markup\Avalonia.Markup.Xaml\PortableXaml\portable.xaml.github\src\Portable.Xaml\Portable.Xaml\XamlObjectWriter.cs:line 586
   at Portable.Xaml.XamlObjectWriterInternal.OnWriteValue(Object value) in C:\Users\Bali\Desktop\Avalonia\src\Markup\Avalonia.Markup.Xaml\PortableXaml\portable.xaml.github\src\Portable.Xaml\Portable.Xaml\XamlObjectWriter.cs:line 574
   at Portable.Xaml.XamlWriterInternalBase.WriteValue(Object value) in C:\Users\Bali\Desktop\Avalonia\src\Markup\Avalonia.Markup.Xaml\PortableXaml\portable.xaml.github\src\Portable.Xaml\Portable.Xaml\XamlWriterInternalBase.cs:line 225
   at Portable.Xaml.XamlObjectWriter.WriteValue(Object value) in C:\Users\Bali\Desktop\Avalonia\src\Markup\Avalonia.Markup.Xaml\PortableXaml\portable.xaml.github\src\Portable.Xaml\Portable.Xaml\XamlObjectWriter.cs:line 240
   at Portable.Xaml.XamlWriter.WriteNode(XamlReader reader) in C:\Users\Bali\Desktop\Avalonia\src\Markup\Avalonia.Markup.Xaml\PortableXaml\portable.xaml.github\src\Portable.Xaml\Portable.Xaml\XamlWriter.cs:line 63
   at Portable.Xaml.XamlServices.Transform(XamlReader xamlReader, XamlWriter xamlWriter, Boolean closeWriter) in C:\Users\Bali\Desktop\Avalonia\src\Markup\Avalonia.Markup.Xaml\PortableXaml\portable.xaml.github\src\Portable.Xaml\Portable.Xaml\XamlServices.cs:line 127
   at Portable.Xaml.XamlServices.Transform(XamlReader xamlReader, XamlWriter xamlWriter) in C:\Users\Bali\Desktop\Avalonia\src\Markup\Avalonia.Markup.Xaml\PortableXaml\portable.xaml.github\src\Portable.Xaml\Portable.Xaml\XamlServices.cs:line 113
   at Avalonia.Markup.Xaml.AvaloniaXamlLoader.LoadFromReader(XamlReader reader, AvaloniaXamlContext context, IAmbientProvider parentAmbientProvider) in C:\Users\Bali\Desktop\Avalonia\src\Markup\Avalonia.Markup.Xaml\AvaloniaXamlLoader.cs:line 207
   at Avalonia.Markup.Xaml.AvaloniaXamlLoader.Load(Stream stream, Assembly localAssembly, Object rootInstance, Uri uri) in C:\Users\Bali\Desktop\Avalonia\src\Markup\Avalonia.Markup.Xaml\AvaloniaXamlLoader.cs:line 186
   at Avalonia.Markup.Xaml.AvaloniaXamlLoader.Load(Type type, Object rootInstance) in C:\Users\Bali\Desktop\Avalonia\src\Markup\Avalonia.Markup.Xaml\AvaloniaXamlLoader.cs:line 92
   at Avalonia.Markup.Xaml.AvaloniaXamlLoader.Load(Object obj) in C:\Users\Bali\Desktop\Avalonia\src\Markup\Avalonia.Markup.Xaml\AvaloniaXamlLoader.cs:line 56
   at ControlCatalog.Pages.CanvasPage.InitializeComponent() in C:\Users\Bali\Desktop\Avalonia\samples\ControlCatalog\Pages\CanvasPage.xaml.cs:line 15
   at ControlCatalog.Pages.CanvasPage..ctor() in C:\Users\Bali\Desktop\Avalonia\samples\ControlCatalog\Pages\CanvasPage.xaml.cs:line 10
   --- End of inner exception stack trace ---
   at System.RuntimeTypeHandle.CreateInstance(RuntimeType type, Boolean publicOnly, Boolean wrapExceptions, Boolean& canBeCached, RuntimeMethodHandleInternal& ctor)
   at System.RuntimeType.CreateInstanceSlow(Boolean publicOnly, Boolean wrapExceptions, Boolean skipCheckThis, Boolean fillCache)
   at System.Activator.CreateInstance(Type type, Boolean nonPublic, Boolean wrapExceptions)
   at Portable.Xaml.Schema.XamlTypeInvoker.CreateInstance(Object[] arguments) in C:\Users\Bali\Desktop\Avalonia\src\Markup\Avalonia.Markup.Xaml\PortableXaml\portable.xaml.github\src\Portable.Xaml\Portable.Xaml.Schema\XamlTypeInvoker.cs:line 265
   at Portable.Xaml.XamlObjectWriterInternal.InitializeObjectIfRequired(Boolean waitForParameters, Boolean required) in C:\Users\Bali\Desktop\Avalonia\src\Markup\Avalonia.Markup.Xaml\PortableXaml\portable.xaml.github\src\Portable.Xaml\Portable.Xaml\XamlObjectWriter.cs:line 773
   at Portable.Xaml.XamlObjectWriterInternal.OnWriteStartObject() in C:\Users\Bali\Desktop\Avalonia\src\Markup\Avalonia.Markup.Xaml\PortableXaml\portable.xaml.github\src\Portable.Xaml\Portable.Xaml\XamlObjectWriter.cs:line 349
   at Portable.Xaml.XamlWriterInternalBase.WriteStartObject(XamlType xamlType) in C:\Users\Bali\Desktop\Avalonia\src\Markup\Avalonia.Markup.Xaml\PortableXaml\portable.xaml.github\src\Portable.Xaml\Portable.Xaml\XamlWriterInternalBase.cs:line 218
   at Portable.Xaml.XamlObjectWriter.WriteStartObject(XamlType xamlType) in C:\Users\Bali\Desktop\Avalonia\src\Markup\Avalonia.Markup.Xaml\PortableXaml\portable.xaml.github\src\Portable.Xaml\Portable.Xaml\XamlObjectWriter.cs:line 229
   at Portable.Xaml.XamlWriter.WriteNode(XamlReader reader) in C:\Users\Bali\Desktop\Avalonia\src\Markup\Avalonia.Markup.Xaml\PortableXaml\portable.xaml.github\src\Portable.Xaml\Portable.Xaml\XamlWriter.cs:line 66
   at Portable.Xaml.XamlServices.Transform(XamlReader xamlReader, XamlWriter xamlWriter, Boolean closeWriter) in C:\Users\Bali\Desktop\Avalonia\src\Markup\Avalonia.Markup.Xaml\PortableXaml\portable.xaml.github\src\Portable.Xaml\Portable.Xaml\XamlServices.cs:line 127
   at Portable.Xaml.XamlServices.Transform(XamlReader xamlReader, XamlWriter xamlWriter) in C:\Users\Bali\Desktop\Avalonia\src\Markup\Avalonia.Markup.Xaml\PortableXaml\portable.xaml.github\src\Portable.Xaml\Portable.Xaml\XamlServices.cs:line 113
   at Avalonia.Markup.Xaml.AvaloniaXamlLoader.LoadFromReader(XamlReader reader, AvaloniaXamlContext context, IAmbientProvider parentAmbientProvider) in C:\Users\Bali\Desktop\Avalonia\src\Markup\Avalonia.Markup.Xaml\AvaloniaXamlLoader.cs:line 207
   at Avalonia.Markup.Xaml.AvaloniaXamlLoader.Load(Stream stream, Assembly localAssembly, Object rootInstance, Uri uri) in C:\Users\Bali\Desktop\Avalonia\src\Markup\Avalonia.Markup.Xaml\AvaloniaXamlLoader.cs:line 186
   at Avalonia.Markup.Xaml.AvaloniaXamlLoader.Load(Type type, Object rootInstance) in C:\Users\Bali\Desktop\Avalonia\src\Markup\Avalonia.Markup.Xaml\AvaloniaXamlLoader.cs:line 92
   at Avalonia.Markup.Xaml.AvaloniaXamlLoader.Load(Object obj) in C:\Users\Bali\Desktop\Avalonia\src\Markup\Avalonia.Markup.Xaml\AvaloniaXamlLoader.cs:line 56
   at ControlCatalog.MainView.InitializeComponent() in C:\Users\Bali\Desktop\Avalonia\samples\ControlCatalog\MainView.xaml.cs:line 34
   at ControlCatalog.MainView..ctor() in C:\Users\Bali\Desktop\Avalonia\samples\ControlCatalog\MainView.xaml.cs:line 14
   --- End of inner exception stack trace ---
   at System.RuntimeTypeHandle.CreateInstance(RuntimeType type, Boolean publicOnly, Boolean wrapExceptions, Boolean& canBeCached, RuntimeMethodHandleInternal& ctor)
   at System.RuntimeType.CreateInstanceSlow(Boolean publicOnly, Boolean wrapExceptions, Boolean skipCheckThis, Boolean fillCache)
   at System.Activator.CreateInstance(Type type, Boolean nonPublic, Boolean wrapExceptions)
   at Portable.Xaml.Schema.XamlTypeInvoker.CreateInstance(Object[] arguments) in C:\Users\Bali\Desktop\Avalonia\src\Markup\Avalonia.Markup.Xaml\PortableXaml\portable.xaml.github\src\Portable.Xaml\Portable.Xaml.Schema\XamlTypeInvoker.cs:line 265
   at Portable.Xaml.XamlObjectWriterInternal.InitializeObjectIfRequired(Boolean waitForParameters, Boolean required) in C:\Users\Bali\Desktop\Avalonia\src\Markup\Avalonia.Markup.Xaml\PortableXaml\portable.xaml.github\src\Portable.Xaml\Portable.Xaml\XamlObjectWriter.cs:line 773
   at Portable.Xaml.XamlObjectWriterInternal.OnWriteStartObject() in C:\Users\Bali\Desktop\Avalonia\src\Markup\Avalonia.Markup.Xaml\PortableXaml\portable.xaml.github\src\Portable.Xaml\Portable.Xaml\XamlObjectWriter.cs:line 349
   at Portable.Xaml.XamlWriterInternalBase.WriteStartObject(XamlType xamlType) in C:\Users\Bali\Desktop\Avalonia\src\Markup\Avalonia.Markup.Xaml\PortableXaml\portable.xaml.github\src\Portable.Xaml\Portable.Xaml\XamlWriterInternalBase.cs:line 218
   at Portable.Xaml.XamlObjectWriter.WriteStartObject(XamlType xamlType) in C:\Users\Bali\Desktop\Avalonia\src\Markup\Avalonia.Markup.Xaml\PortableXaml\portable.xaml.github\src\Portable.Xaml\Portable.Xaml\XamlObjectWriter.cs:line 229
   at Portable.Xaml.XamlWriter.WriteNode(XamlReader reader) in C:\Users\Bali\Desktop\Avalonia\src\Markup\Avalonia.Markup.Xaml\PortableXaml\portable.xaml.github\src\Portable.Xaml\Portable.Xaml\XamlWriter.cs:line 66
   at Portable.Xaml.XamlServices.Transform(XamlReader xamlReader, XamlWriter xamlWriter, Boolean closeWriter) in C:\Users\Bali\Desktop\Avalonia\src\Markup\Avalonia.Markup.Xaml\PortableXaml\portable.xaml.github\src\Portable.Xaml\Portable.Xaml\XamlServices.cs:line 127
   at Portable.Xaml.XamlServices.Transform(XamlReader xamlReader, XamlWriter xamlWriter) in C:\Users\Bali\Desktop\Avalonia\src\Markup\Avalonia.Markup.Xaml\PortableXaml\portable.xaml.github\src\Portable.Xaml\Portable.Xaml\XamlServices.cs:line 113
   at Avalonia.Markup.Xaml.AvaloniaXamlLoader.LoadFromReader(XamlReader reader, AvaloniaXamlContext context, IAmbientProvider parentAmbientProvider) in C:\Users\Bali\Desktop\Avalonia\src\Markup\Avalonia.Markup.Xaml\AvaloniaXamlLoader.cs:line 207
   at Avalonia.Markup.Xaml.AvaloniaXamlLoader.Load(Stream stream, Assembly localAssembly, Object rootInstance, Uri uri) in C:\Users\Bali\Desktop\Avalonia\src\Markup\Avalonia.Markup.Xaml\AvaloniaXamlLoader.cs:line 186
   at Avalonia.Markup.Xaml.AvaloniaXamlLoader.Load(Type type, Object rootInstance) in C:\Users\Bali\Desktop\Avalonia\src\Markup\Avalonia.Markup.Xaml\AvaloniaXamlLoader.cs:line 92
   at Avalonia.Markup.Xaml.AvaloniaXamlLoader.Load(Object obj) in C:\Users\Bali\Desktop\Avalonia\src\Markup\Avalonia.Markup.Xaml\AvaloniaXamlLoader.cs:line 56
   at ControlCatalog.MainWindow.InitializeComponent() in C:\Users\Bali\Desktop\Avalonia\samples\ControlCatalog\MainWindow.xaml.cs:line 25
   at ControlCatalog.MainWindow..ctor() in C:\Users\Bali\Desktop\Avalonia\samples\ControlCatalog\MainWindow.xaml.cs:line 12
   --- End of inner exception stack trace ---
   at System.RuntimeTypeHandle.CreateInstance(RuntimeType type, Boolean publicOnly, Boolean wrapExceptions, Boolean& canBeCached, RuntimeMethodHandleInternal& ctor)
   at System.RuntimeType.CreateInstanceSlow(Boolean publicOnly, Boolean wrapExceptions, Boolean skipCheckThis, Boolean fillCache)
   at System.Activator.CreateInstance[T]()
   at Avalonia.Controls.AppBuilderBase`1.Start[TMainWindow](Func`1 dataContextProvider) in C:\Users\Bali\Desktop\Avalonia\src\Avalonia.Controls\AppBuilderBase.cs:line 125
   at ControlCatalog.NetCore.Program.Main(String[] args) in C:\Users\Bali\Desktop\Avalonia\samples\ControlCatalog.NetCore\Program.cs:line 33
```

### Attempts to Resolve

1. Manually copy `libSkiaSharp.dll` into the bin folder. Did not work.

### Related Issue

https://github.com/zkSNACKs/WalletWasabi/issues/696

**Comment by danwalmsley** [comment]
Possibly need to copy VC++ runtime dependencies...

https://github.com/wieslawsoltes/Core2D/blob/master/build.cake#L76

**Comment by nopara73** [comment]
Yes, it resolved the issue: https://www.microsoft.com/en-us/download/details.aspx?id=52685

**Comment by drearyrainDeng** [comment]
can not run on win10 ! (version 0.7.0) miss libSkiaSharp.dll  !

**Comment by nopara73** [comment]
@drearyrainDeng This issue has been resolved by Avalonia itself, thus our hack is not needed anymore. We tested it. While I suspect you are using an older version of Avalonia than we do I reopen this issue to get to the bottom of this, because it is important.  

My theory:  
We're using Avalonia in a strange unconventional way, this makes sure we can update to the latest at all times. However @drearyrainDeng is using from NuGet, thus he doesn't yet have the fix. @danwalmsley Am I correct?
